functions/model/tasks/tark/deleteTask.js

```diff
@@ -119,7 +119,13 @@ exports.deleteTask = function(request, response) {
                 console.log("Error:", error);
             });
 
-            const promises = [p1, p2, p3, p4];
+            const sendMailToWatcherPromise = getTask(taskId, orgDomain).then((taskDoc) => {
+                taskDoc.Watcher.forEach(element => {
+                    watchers.push(element);
+                });
+            });
+
```

Comment:
No need for these changes, Also the logic can be improved. For the time being please remove it.

Reply:
removed